src/track/track.h

```diff
@@ -283,16 +283,17 @@ class Track : public QObject {
     // Get the track's Beats list
     mixxx::BeatsPointer getBeats() const;
 
-    // Set the track's Beats
-    void setBeats(mixxx::BeatsPointer beats);
+    // Set the track's Beats if not locked
+    bool trySetBeats(mixxx::BeatsPointer beats, bool lockBpmAfterSet);
```

Comment:
lockBpmAfterSet = false to prevent repeating this parameter for >90% of all invocations. Why bother the majority of use cases with a parameter that is only set to true for some special case? Passing false is both the common use case and safe variant -> default value.